genic/main.c

```diff
@@ -46,19 +56,44 @@ int main(int argc, char **argv)
       endrun(0, "%s\n", big_file_get_error_message());
   }
   saveheader(&bf, TotNumPart);
+  /*Use 'total' (CDM + baryon) transfer function
+   * unless DifferentTransferFunctions are on.
+   * Note that massive neutrinos, if present,
+   * will be followed elsewhere.*/
+  int DMType = 2, GasType = 2;
+  if(ProduceGas && DifferentTransferFunctions) {
+      DMType = 1;
+      GasType = 0;
+  }
 
   /*First compute and write CDM*/
-  displacement_fields(1);
+  displacement_fields(DMType);
   write_particle_data(1, &bf);
   free_ffts();
 
   /*Now make the gas if required*/
   if(ProduceGas) {
     setup_grid(shift_gas, TotNumPart);
-    displacement_fields(0);
+    displacement_fields(GasType);
     write_particle_data(0, &bf);
     free_ffts();
   }
+  /*Now add random velocity neutrino particles*/
+  if(NGridNu > 0) {
+      const int64_t TotNu = (int64_t) NGridNu*NGridNu*NGridNu;
+      const double kBMNu = 3*CP.ONu.kBtnu / (CP.MNu[0]+CP.MNu[1]+CP.MNu[2]);
+      const double v_th = NU_V0(InitTime, kBMNu, UnitVelocity_in_cm_per_s);
+      const double total_frac = init_thermalvel(v_th, Max_nuvel/v_th, 0);
+      int i;
+      message(0,"F-D velocity scale: %g. Max particle vel: %g. Fraction of mass in particles: %g\n",v_th*sqrt(InitTime), Max_nuvel*sqrt(InitTime), total_frac);
+      setup_grid(shift_nu, TotNu);
+      for(i = 0; i< TotNu; i++)
+          add_thermal_speeds(P[i].Vel);
```

Comment:
Is this Peculiar (a dx /dt ) or a**2 dx / dt

Reply:
Peculiar a dx/dt, I think.

Reply:
You have to be careful, because the unit of P[i].Vel depends on whether UsePecuiliar Velocity is set.

Reply:
Well noticed - I had not accounted for that. Should be fixed now.